README.md

```diff
@@ -115,123 +113,117 @@ GoogleSignin.configure({
+#### - `signInSilently()`
 
-Obtain the user access token.
+May be called eg. in the `componentDidMount` of your main component. This method returns the [current user](#3-userinfo) if they already signed in and `null` otherwise.
+
+To see how to handle errors read [`signIn()` method](#--signin)
 
 ```js
-GoogleSignin.getAccessToken()
-  .then(token => {
-    console.log(token);
-  })
-  .catch(err => {
-    console.log(err);
-  });
+getCurrentUser = async () => {
+  try {
+    const userInfo = await GoogleSignin.signInSilently();
+    this.setState({ userInfo });
+  } catch (error) {
+    console.error(error);
+  }
+};
 ```
 
-#### - signOut
+#### - `signOut()`
 
-remove user session from the device
+Remove user session from the device.
 
 ```js
-_signOut = async () => {
+signOut = async () => {
   try {
     await GoogleSignin.revokeAccess();
     await GoogleSignin.signOut();
-    this.setState({ user: null });
+    this.setState({ user: null }); // Remember to remove the user from your app's state as well
   } catch (error) {
-    this.setState({
-      error,
-    });
+    console.error(error)
   }
 };
 ```
 
-#### - revokeAccess
+#### - `revokeAccess()`
 
-remove your application from the user authorized applications
+Remove your application from the user authorized applications.
 
 ```js
 GoogleSignin.revokeAccess()
   .then(() => {
     console.log('deleted');
   })
-  .catch(err => {
-    console.log(err);
+  .catch(error => {
+    console.error(error);
   });
 ```
 
-### 3. User
+#### - `statusCodes`
+
+These are useful when determining which kind of error has occured during sign in process. Import `statusCodes` along with `GoogleSignIn`. Under the hood these constants are derived from native GoogleSignIn error codes and are platform specific and can be changed. Always prefer to compare `error.code` to `statusCodes.SIGN_IN_CANCELLED` or `statusCodes.IN_PROGRESS` and not relying on raw value of the `error.code`.
+
+Name | Description 
+--- | ---
+`SIGN_IN_CANCELLED` | When user cancels the sign in flow
+`IN_PROGRESS` | Trying to invoke another sign in flow when previous one has not yet finished
+
+[Example how to use `statusCodes`](#--signin).
 
-This is the typical information you obtain once the user sign in:
+
+### 3. `userInfo`
+
+Example `userInfo` which is returned after successful sign in.
 
 ```
-  {
-    id: <user id. do not use on the backend>
-    name: <user name>
-    givenName: <user given name> (Android only)
-    familyName: <user family name> (Android only)
-    email: <user email>
-    photo: <user picture profile>
-    idToken: <token to authenticate the user on the backend>
-    serverAuthCode: <one-time token to access Google API from the backend on behalf of the user>
-    scopes: <list of authorized scopes>
-    accessToken: <needed to access google API from the application>
+{
+  idToken: string,
+  accessToken: string | null,
+  accessTokenExpirationDate: number | null, // timestamp, on Android this is null 
```

Comment:
`@"accessTokenExpirationDate": [NSNumber numberWithDouble:user.authentication.accessTokenExpirationDate.timeIntervalSinceNow]` - I think we should not be calling it a timestamp since it's a timeIntervalSinceNow. This also makes me wonder if it should be a timeIntervalSinceNow..?

Reply:
Let's make it timestamp. Also I would like to know if it's milliseconds or seconds.

Reply:
We decided to deprecate `accessTokenExpirationDate` since it is quite problematic in many ways but mainly it is not available on Android.